### ISSUE TITLE
Am having very low accuracies while training

### ISSUE THREAD

**Issue by junhuplim** [open-mmlab/mmclassification#469]
### Checklist
- I have searched related issues but cannot get the expected help.
- I have read related documents and don't know what to do.

### Describe the question you meet
I need help to ensure that I am not doing anything wrong, as I am getting rather low values on my accuracies while training a VGG for classification even though I am using the pretrained weights.  My training set contains of 786 classes (logos)- where some classes might only contain <10 images. 

### Post related information
1. The output of `pip list | grep "mmcv\|mmcls\|^torch"`
```python
torch                             1.7.1+cu101
torchaudio                        0.7.0
torchfile                         0.1.0
torchnet                          0.0.4
torchsummary                      1.5.1
torchvision                       0.8.2
```
2. Your config file if you modified it or created a new one.
```python
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='VGG', depth=16, norm_cfg=dict(type='BN'), num_classes=786),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    )
)

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)

train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop', size=224, backend='pillow'),
    dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', size=(256, -1), backend='pillow'),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]

data = dict(
    samples_per_gpu=16,
    workers_per_gpu=2,
    train=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/train',
        pipeline=train_pipeline
    ),
    val=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/val',
        ann_file='/workspace/jupyter_notebooks/datasets/mmclass/classification/val.txt',
        pipeline=test_pipeline
    ),
    test=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/val',
        ann_file='/workspace/jupyter_notebooks/datasets/mmclass/classification/val.txt',
        pipeline=test_pipeline
))

evaluation = dict(interval=1, metric='accuracy')
optimizer = dict(type='SGD', lr=0.00125, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)

lr_config = dict(policy='step', step=[30, 60, 90])
runner = dict(type='EpochBasedRunner', max_epochs=100)

checkpoint_config = dict(interval=1)
log_config = dict(
    interval=500,
    hooks=[
        dict(type='TextLoggerHook'),
    ])

dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'load/vgg16_bn_batch256_imagenet_20210208-7e55cd29.pth'
resume_from = None
workflow = [('train', 1)]
    
```
3. Your train log file if you meet the problem during training.
accuracy_top-1 is always 0 and top-5 is < 1.
```python
2021-09-28 07:36:18,798 - mmcls - INFO - Epoch [10][500/1439]	lr: 1.250e-03, eta: 2:16:35, time: 0.090, data_time: 0.007, memory: 3915, loss: 1.4455
2021-09-28 07:37:01,614 - mmcls - INFO - Epoch [10][1000/1439]	lr: 1.250e-03, eta: 2:17:50, time: 0.086, data_time: 0.003, memory: 3915, loss: 1.4632
2021-09-28 07:37:39,205 - mmcls - INFO - Saving checkpoint at 10 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>] 2296/2296, 319.8 task/s, elapsed: 7s, ETA:     0s2021-09-28 07:37:52,417 - mmcls - INFO - Epoch(val) [10][144]	accuracy_top-1: 0.0000, accuracy_top-5: 0.5662
```
4. Other code you modified in the `mmcls` folder.

**Comment by Ezra-Yu** [comment]
the value of `accuracy_top-5` is [0-1].
 
 1.  you can modify your fine-tune script refer to [the new doc](https://mmclassification--466.org.readthedocs.build/en/466/tutorials/finetune.html) in #466 .
 2.  check your dataset, if there is some wrong with the label info.

**Comment by mzr1996** [comment]
Hello, I create a config based on your config and change the dataset to CIFAR-10.
Please try to train this config and report the training log, it will help us to locate the problem.
```python
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='VGG', depth=16, norm_cfg=dict(type='BN'), num_classes=10),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    )
)

img_norm_cfg = dict(
    mean=[125.307, 122.961, 113.8575], std=[51.5865, 50.847, 51.255], to_rgb=False,)

train_pipeline = [
    dict(type='RandomCrop', size=32, padding=4),
    dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(type='Resize', size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]

test_pipeline = [
    dict(type='Resize', size=(256, -1), backend='pillow'),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]

data = dict(
    samples_per_gpu=16,
    workers_per_gpu=2,
    train=dict(
        type='CIFAR10',
        data_prefix='data/cifar10',
        pipeline=train_pipeline),
    val=dict(
        type='CIFAR10',
        data_prefix='data/cifar10',
        pipeline=test_pipeline,
        test_mode=True),
    test=dict(
        # replace `data/val` with `data/test` for standard test
        type='CIFAR10',
        data_prefix='data/cifar10',
        pipeline=test_pipeline,
        test_mode=True
))

evaluation = dict(interval=1, metric='accuracy')
optimizer = dict(type='SGD', lr=0.00125, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)

lr_config = dict(policy='step', step=[30, 60, 90])
runner = dict(type='EpochBasedRunner', max_epochs=100)

checkpoint_config = dict(interval=1)
log_config = dict(
    interval=500,
    hooks=[
        dict(type='TextLoggerHook'),
    ])

dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'load/vgg16_bn_batch256_imagenet_20210208-7e55cd29.pth'
resume_from = None
workflow = [('train', 1)]
```

**Comment by junhuplim** [comment]
@mzr1996 I have restarted training with your config and it gives good accuracy (even at first epoch) 
```python
[>>>>>>>>>>>>>>>>>>>>>>>>] 10000/10000, 539.8 task/s, elapsed: 19s, ETA:     0s2021-09-28 08:57:44,692 - mmcls - INFO - Epoch(val) [1][625]	accuracy_top-1: 90.7100, accuracy_top-5: 99.7200
``` 

This probably mean there is some problem with my dataset again? May I ask what are the things I should be looking out for when building my own custom dataset? I have mainly referred to the docs and reformatted my dataset to follow the ImageNet format.

**Comment by mzr1996** [comment]
> @Ezra-Yu @mzr1996 I have restarted training with your config and it gives good accuracy (even at first epoch)
> 
> ```python
> [>>>>>>>>>>>>>>>>>>>>>>>>] 10000/10000, 539.8 task/s, elapsed: 19s, ETA:     0s2021-09-28 08:57:44,692 - mmcls - INFO - Epoch(val) [1][625]	accuracy_top-1: 90.7100, accuracy_top-5: 99.7200
> ```
> 
> This probably mean there is some problem with my dataset? May I ask what are the things I should be looking out for when building my own custom dataset? I have mainly referred to the docs and reformatted my dataset to follow the ImageNet format.
> 
> I do have classes with only <10 images per class, while other classes might have more images in them. Also, their image size (can be as small as 24x24) is much smaller than 224x224. Since VGG resizes them to 224x224, I imagine the image to be upscaled rather significantly. I wonder if this affects too.

Hello, I think that's because of the wrong class mapping. Please delete the `ann_file` field in your val and test dataset config. And reorganize your test data folder as the train data folder.

**Comment by junhuplim** [comment]
Okay, I have tried using the same format for my val and test. My current config:

```python 
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='VGG', depth=16, norm_cfg=dict(type='BN'), num_classes=786),
    neck=None,
    head=dict(
        type='ClsHead',
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=(1, 5),
    )
)

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)

train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop', size=224, backend='pillow'),
    dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', size=(256, -1), backend='pillow'),
#     dict(type='Resize', size=(224, 224), backend='pillow'),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]

data = dict(
    samples_per_gpu=16,
    workers_per_gpu=2,
    train=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/train',
        pipeline=train_pipeline
    ),
    val=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/val',
        pipeline=test_pipeline,
        test_mode=True
    ),
    test=dict(
        type='ImageNet',
        data_prefix='/workspace/jupyter_notebooks/datasets/mmclass/classification/val',
        pipeline=test_pipeline,
        test_mode=True
    ),
)

evaluation = dict(interval=1, metric='accuracy')
optimizer = dict(type='SGD', lr=0.00125, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)

lr_config = dict(policy='step', step=[30, 60, 90])
runner = dict(type='EpochBasedRunner', max_epochs=10)

checkpoint_config = dict(interval=1)
log_config = dict(
    interval=500,
    hooks=[
        dict(type='TextLoggerHook'),
    ])

dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'load/vgg16_bn_batch256_imagenet_20210208-7e55cd29.pth'
resume_from = None
workflow = [('train', 1)]
 ```

I am still getting the same error of low accuracy. However, when I tried to use my `train dataset` for my validation, accuracy is high and normal. Question now:

If i have 700 classes in training set, do I also need 700 classes in validation set? I am asking this as I have removed the classes with no images in my validation set (since there might only be 1 image in the training set), resulting in a lower number of classes in my validation set (say 300+).
